waku/discovery/waku_discv5.nim

```diff
@@ -203,6 +203,7 @@ proc searchLoop(wd: WakuDiscoveryV5) {.async.} =
     let discoveredPeers =
       discoveredRecords.mapIt(it.toRemotePeerInfo()).filterIt(it.isOk()).mapIt(it.value)
 
+    trace "discv5 discovered peers", num_discovered_peers = discoveredPeers.len
```

Comment:
Can we also add trace logs as to if a peer was filtered out, and if so the reason for it?

Reply:
And also log peerIDs that were discovered

Reply:
Thanks for the comments @chaitanyaprem !
In this commit, https://github.com/waku-org/nwaku/pull/2705/commits/8a231621575f0e358c9f1a74e5849354d40f0eb9, we add `trace` logs that can show info similar to the following:
```code
TRC 2024-05-16 17:39:33.132+02:00 discv5 discovered peers                    topics="waku discv5" tid=2309898 file=waku_discv5.nim:216 num_discovered_peers=22 peers="@[\"16U*4TZFpq\", \"16U*3ixik1\", \"16U*1Ttos1\", \"16U*V8fdva\", \"16U*WUFKT3\", \"16U*YUFDcd\", \"16U*3beNzJ\", \"16U*pCkGbJ\", \"16U*muYdZr\", \"16U*xgrftw\", \"16U*TLZys5\", \"16U*3Zqsge\", \"16U*2axTsD\", \"16U*G2c9HT\", \"16U*Bz9iJB\", \"16U*pASYP9\", \"16U*EndhSR\", \"16U*jvUfsU\", \"16U*wsmTgW\", \"16U*Gq3kcc\", \"16U*NVwNvv\", \"16U*hV6jh3\"]"
TRC 2024-05-16 17:39:33.132+02:00 discv5 discarded wrong records             topics="waku discv5" tid=2309898 file=waku_discv5.nim:220 wrong_records=@[]
```

```code
TRC 2024-05-16 17:40:08.291+02:00 discv5 discovered peers                    topics="waku discv5" tid=2309898 file=waku_discv5.nim:216 num_discovered_peers=87 peers="@[\"16U*u4mh4q\", \"16U*CjTL4s\", \"16U*9t9a8K\", \"16U*FcpHak\", \"16U*PCqRLn\", \"16U*cwCkGe\", \"16U*2pCHE1\", \"16U*D6ej5W\", \"16U*ox8mEy\", \"16U*QXfjdB\", \"16U*BWRJLo\", \"16U*c4DfbB\", \"16U*CjwmpV\", \"16U*1c3zbn\", \"16U*haritG\", \"16U*wr6UUP\", \"16U*mMpKzj\", \"16U*beQ7sf\", \"16U*PP8HsP\", \"16U*JfBfd8\", \"16U*VSq769\", \"16U*r46Eau\", \"16U*Cb6Z9j\", \"16U*hx3ShF\", \"16U*zMcJ81\", \"16U*p1DKzT\", \"16U*RSLFeH\", \"16U*iCBf2E\", \"16U*azcR8U\", \"16U*vkCdqF\", \"16U*6niMvq\", \"16U*wLpVpt\", \"16U*kAcXJF\", \"16U*ood9Q8\", \"16U*LtRknW\", \"16U*eWKACF\", \"16U*SqHmeq\", \"16U*Gk72f2\", \"16U*MVZwT2\", \"16U*9gtFQb\", \"16U*7kaH8U\", \"16U*8Jgd8Y\", \"16U*AFNkuB\", \"16U*9sk5Ma\", \"16U*M2nkNJ\", \"16U*wAXHym\", \"16U*SJjUWu\", \"16U*zXbcGe\", \"16U*vg9XwR\", \"16U*miLKLo\", \"16U*qc3Gta\", \"16U*Ach9rb\", \"16U*wV3GBL\", \"16U*bPcdHq\", \"16U*2L4zHz\", \"16U*kJV3DP\", \"16U*MjPbzs\", \"16U*SSW45X\", \"16U*qGFpE6\", \"16U*73e3Uy\", \"16U*BL3HtK\", \"16U*zhnd17\", \"16U*SFd3z9\", \"16U*eC78BU\", \"16U*P1ei4u\", \"16U*EenMkk\", \"16U*22T7Wm\", \"16U*N3yadf\", \"16U*FscA3x\", \"16U*pP8v6T\", \"16U*GMEtcw\", \"16U*31cbtz\", \"16U*T8pVg8\", \"16U*vZVb4K\", \"16U*Nu5LQa\", \"16U*2QJqFJ\", \"16U*PVKc4V\", \"16U*7iz2KN\", \"16U*FPUEYC\", \"16U*fqo8tJ\", \"16U*bgv6zF\", \"16U*YnoKBp\", \"16U*U5BPVT\", \"16U*urCguJ\", \"16U*uFFUC7\", \"16U*cMm3UC\", \"16U*vxUm4w\"]"
TRC 2024-05-16 17:40:08.292+02:00 discv5 discarded wrong records             topics="waku discv5" tid=2309898 file=waku_discv5.nim:220 wrong_records=@[]
```